src/amr_criteria/amr_criteria.cpp

```diff
@@ -95,6 +95,26 @@ AmrTag AMRFirstDerivative::operator()(const MeshBlockData<Real> *rc) const {
   return Refinement::FirstDerivative(bnds, q, refine_criteria, derefine_criteria);
 }
 
+void AMRFirstDerivative::operator()(MeshData<Real> *mc,
+                                    ParArray1D<AmrTag> &delta_levels) const {
+  auto ib = mc->GetBoundsI(IndexDomain::interior);
+  auto jb = mc->GetBoundsJ(IndexDomain::interior);
+  auto kb = mc->GetBoundsK(IndexDomain::interior);
+  auto bnds = AMRBounds(ib, jb, kb);
+  auto dims = mc->GetMeshPointer()->resolved_packages->FieldMetadata(field).Shape();
+  int n5(0), n4(0);
+  if (dims.size() > 2) {
+    n5 = dims[1];
+    n4 = dims[2];
+  } else if (dims.size() > 1) {
+    n5 = dims[0];
+    n4 = dims[1];
+  }
+  const int idx = comp4 + n4 * (comp5 + n5 * comp6);
```

Comment:
I'm wondering if we don't have a simpler way to get a flat index (I imagine there must be more places with thus use case but I don't recall offhand).